**1.Статистические гипотезы**

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (использовать двусторонний тест)
Указания к решению задачи:

Указать какой тест применяется в задаче
Создать массив, заполненный указанными значениями.
Решение задачи должно быть представлено в виде кода с выводом : среднего значения, дисперсии, стандартного отклонения и значения критерия.
По таблице определить вручную границы доверительного интервала.
Сравнить ответ со значением границ.
Добавить в программу следующие строки
t_t = stats.t.ppf(1.0-alfa/2, df)
stats.ttest_1samp(x,mu_0)
Что является результатом вызова данной функции?

In [ ]:
import numpy as np
n = 10                                 #количество пачек
alpha = 0.01                           #доверительная вероятность равна 1-0.99
mu_0=200                               #начальное среднее
вес = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190]) #создание массива, заполненного весом каждой пачки
среднее = вес.mean()                   #среднее значение 
СКО = вес.std(ddof=1)                  #среднеквадратическое откл.
z = (среднее - mu_0) / (СКО / n**(1/2))#z-критерий
t = 3.25                               #значение t-критерия по таблице (0.01 и 9)
print(z)
print(t)
print(mu_0)
(среднее - t * СКО / np.sqrt(n), среднее + t * СКО / np.sqrt(n))


-1.0651074037450896
3.25
200


(193.9229967955732, 203.0770032044268)

Значение попадает в доверительный интервал.
Решение выполнено с помощью двустороннего теста Стьюдента.

Ответ: утверждение продавца верно.

In [ ]:
from scipy import stats
t_t = stats.t.ppf(1-alpha/2, 9)
print(t_t)                          #значение t-критерия
stats.ttest_1samp(вес,mu_0)         #критерий и p-значение

3.2498355440153697


Ttest_1sampResult(statistic=-1.0651074037450896, pvalue=0.31456660352472854)

Результатом данной функции будет являться вывод информации t-теста для среднего значения группы оценок.
Это проверка нулевой гипотезы о том, что ожидаемое значение (среднее) выборки независимых наблюдений  равно заданному среднему значению. 
 p-значение = 0.314 не ниже 0,01 (доверительная вероятность равна 99%), поэтому мы не можем отвергнуть нулевую гипотезу. 



---



**2.Доверительные интервалы**

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 

6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 

Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.


Указания к решению задачи:


Создать массив, заполненный указанными значениями.
Решение задачи должно быть представлено в виде кода с выводом: среднего значения, дисперсии, стандартного отклонения, правой и левой границ интервала.
По таблице определить вручную границы доверительного интервала.
Сравнить ответ со значением границ.


In [ ]:
X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n_X = 10                                         #число измерений
alpha_X = 0.05                                   #доверительная вероятность 1-95%
среднее_X = X.mean()                             #среднее
дисперсия_X = X.var()                            #дисперсия
СКО_X = X.std(ddof=1)                            #среднеквадратическое откл
t_X = stats.t(n_X - 1).ppf(1 - alpha_X/2)        #t-критерий
границы = (среднее_X - t_X * СКО_X/np.sqrt(n_X), среднее_X + t_X * СКО_X/np.sqrt(n_X))
границы

(6.267515851415713, 6.912484148584288)

Ответ: истинное значение X с доверительной вероятностью 95% лежит в пределах от 6.267515851415713 до 6.912484148584288

---



**3. Плотность распределения вероятностей (указать только ответы)**

Рост взрослого населения города X имеет нормальное распределение. Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см. Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:

а). больше 182 см

б). больше 190 см

в). от 166 см до 190 см

г). от 166 см до 182 см

д). от 158 см до 190 см

е). не выше 150 см или не ниже 190 см

ё). не выше 150 см или не ниже 198 см

ж). ниже 166 см.
 


In [ ]:
#а) 16.04%
#б) 4.56%
#в) 81.8%
#г) 68.26%
#д) 95.44%
#е) 2.5%
#ё) 0.4%
#ж) 16.04%